Jenkinsfile

```diff
@@ -345,8 +345,13 @@ def archiveTestOutput(Map args = [:]) {
     }
     cmd(label: 'Prepare test output', script: 'python .ci/scripts/pre_archive_test.py')
     dir('build') {
+      if (isUnix()) {
+        cmd(label: 'Delete folders that are causing exceptions (See JENKINS-58421)',
+            returnStatus: true,
+            script: 'rm -rf ve || true; find . -type d -name vendor -exec rm -r {} \\;')
+      } else { log(level: 'INFO', text: 'Delete folders that are causing exceptions (See JENKINS-58421) is disabled for Windows.') }
       junitAndStore(allowEmptyResults: true, keepLongStdio: true, testResults: args.testResults, stashedTestReports: stashedTestReports, id: args.id)
-      archiveArtifacts(allowEmptyArchive: true, artifacts: args.artifacts)
+      tar(file: "test-build-artifacts-${args.id}.tgz", dir: '.', archive: true, allowMissing: true)
```

Comment:
This might help us to avoid the interruption error even when the cleanup defined above happened:

```
[2020-10-05T12:09:17.924Z] drwxr-xr-x  26 jenkins jenkins 4.0K Oct  5 12:09 ..
[2020-10-05T12:09:17.924Z] + find . -name build -ls
[2020-10-05T12:09:17.924Z]   5170689      4 drwxr-xr-x   4 jenkins  jenkins      4096 Oct  5 12:09 ./metricbeat/build
[2020-10-05T12:09:17.924Z]   5168179      4 drwxr-xr-x   3 jenkins  jenkins      4096 Oct  5 11:39 ./ve/docker/lib/python3.7/site-packages/pip/_internal/operations/build
[2020-10-05T12:09:17.934Z] [Pipeline] isUnix
[2020-10-05T12:09:17.937Z] [Pipeline] sh
[2020-10-05T12:09:18.224Z] + rm -rf ve
[2020-10-05T12:09:18.224Z] + find . -type d -name vendor -exec rm -r {} ;
[2020-10-05T12:09:18.239Z] [Pipeline] junit
[2020-10-05T12:09:18.239Z] WARNING: Unknown parameter(s) found for class type 'hudson.tasks.junit.pipeline.JUnitResultsStep': id,stashedTestReports
[2020-10-05T12:09:18.242Z] Recording test results
[2020-10-05T12:09:19.975Z] [Pipeline] stash
[2020-10-05T12:09:20.208Z] Stashed 1 file(s)
[2020-10-05T12:09:20.213Z] [Pipeline] archiveArtifacts
[2020-10-05T12:09:20.219Z] Archiving artifacts
[2020-10-05T12:09:20.450Z] java.lang.InterruptedException: no matches found within 10000
[2020-10-05T12:09:20.450Z] 	at hudson.FilePath$ValidateAntFileMask.hasMatch(FilePath.java:2865)
[2020-10-05T12:09:20.450Z] 	at hudson.FilePath$ValidateAntFileMask.invoke(FilePath.java:2744)
[2020-10-05T12:09:20.450Z] 	at hudson.FilePath$ValidateAntFileMask.invoke(FilePath.java:2725)
[2020-10-05T12:09:20.450Z] 	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3112)
[2020-10-05T12:09:20.450Z] Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from beats-ci-immutable-ubuntu-1804-1601897573326600637.c.elastic-ci-prod.internal/10.224.1.213:57106
```